extensions/machine-learning-services/src/common/apiWrapper.ts

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+
+/**
+ * Wrapper class to act as a facade over VSCode and Data APIs and allow us to test / mock callbacks into
+ * this API from our code
+ */
+export class ApiWrapper {
```

Comment:
If you are going to be using the vscode test runner you don't really need this wrapper. This if only if you need to mock the vscode apis for some reason.

Reply:
I'm using this to mock vs code and azdata functions 

---
In reply to: [357335445](https://github.com/microsoft/azuredatastudio/pull/8622#discussion_r357335445) [](ancestors = 357335445)